furnace.c

```diff
@@ -230,6 +258,39 @@ command_handler(furnace_context_t* ctx, uint8_t* buffer, void (*feedback)(const
     }
   }
 #endif
+#if CONFIG_WATER
+  else if(sscanf(buffer, "water %u", &arg) == 1) {
+    const int pwm_value = arg + WATER_OFFSET;
+    if(arg < 0){
+      const char msg[] = "water pwm argument too small!\r\n";
+      const size_t msg_len = sizeof(msg)-1;
+      feedback(msg, msg_len);
+    }else if(arg > 10){
+      const char msg[] = "water pwm argument too big!\r\n";
+      const size_t msg_len = sizeof(msg)-1;
+      feedback(msg, msg_len);
+    } else {
+    const int res = set_pwm_safe(WATER_PIN, ctx, pwm_value);
+      if (res  == 1){
+        const char msg[] = "water pwm argument too big!\r\n";
+        const size_t msg_len = sizeof(msg)-1;
+        feedback(msg, msg_len);
+      } else if(res == -1) {
+        const char msg[] = "set_pwm_safe: unexpected pin argument!\r\n";
+        const size_t msg_len = sizeof(msg)-1;
+        feedback(msg, msg_len);
+      } else if(res != 0) {
+        const char msg[] = "set_pwm_safe: unexpected return value!\r\n";
+        const size_t msg_len = sizeof(msg)-1;
+        feedback(msg, msg_len);
+      }
+    }
+  } else if(memcmp(buffer, "water\n", 6) == 0){
+    char msg[16];
+    const size_t msg_len = snprintf(msg, sizeof(msg), "water = %d\r\n", ctx->pwm_water);
+    feedback(msg, msg_len);
+  }
```

Comment:
At his point, It's probably more clear to move some of it this to a separate function and do an early return instead of some many if/else statements. Something like:
```C
void handle_command_water(ctx, feedback, user_pwm_water)
{
  if (user_pwm_water > 10) {
    /* Report user interface error */
    return;
  }

  pwm_value = user_pwm_water + WATER_OFFSET;
  res = set_pwm_safe(WATER_PIN, ctx, pwm_value);

  if (res == 0)
    return;

  if (res == ...) {
    /* Report other errors */
  }
}

/* ... */

  else if(sscanf(buffer, "water %u", &arg) == 1) {
    handle_command_water(ctx, feedback, arg);
  } else if(memcmp(buffer, "water\n", 6) == 0){
    char msg[16];
    const size_t msg_len = snprintf(msg, sizeof(msg), "water = %d\r\n", ctx->pwm_water);
    feedback(msg, msg_len);
  }
```